ktor-features/ktor-serialization/jvm/src/io/ktor/serialization/SerializationConverter.kt

```diff
@@ -110,16 +110,39 @@ private constructor(
         contentType: ContentType,
         value: Any
     ): Any? {
-        @Suppress("UNCHECKED_CAST")
-        val serializer = serializerForSending(context, value, format.serializersModule) as KSerializer<Any>
+        val result = try {
+            serializerFromResponseType(context, format.serializersModule)?.let {
+                serializeContent(it, format, value, contentType, context)
+            }
+        } catch (cause: SerializationException) {
+            // can fail due to
+            // 1. https://github.com/Kotlin/kotlinx.serialization/issues/1163)
+            // 2. mismatching between compile-time and runtime types of the response.
+            null
+        }
+        if (result != null) {
+            return result
+        }
 
+        val manualSearchSerializer = searchSerializer(value, format.serializersModule)
```

Comment:
Manual sounds like we have a human pool resolving every particular case :) The search function is trying to find or guess serializer in some cases. So I would name this thinking of this "guessing"